test/TimerGPIOtask_test.rb

```diff
@@ -41,6 +41,17 @@
                    "expected the true state to be sent for at least 1.5 seconds, "\
                    "but got #{toc - tic}"
         end
+
+        it "it sets default value to gpio in case of stop hook" do
+            @task = create_configure_and_start_task(duration: 2)
+
+            expect_execution { task.stop! }
+                .join_all_waiting_work(false)
+                .to do
+                    have_one_new_sample(task.gpio_state_port)
+                        .matching { |v| v.states[0].data == 0 }
+                end
```

Comment:
I you're not writing the result, the test teardown should be complaining (plus, you should really be checking that the task waits for feedback before it stops)